src/main/java/ch/njol/skript/conditions/CondIsDivisibleBy.java

```diff
@@ -0,0 +1,64 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Checker;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Divisible By")
+@Description("Check if a number is divisible by another number.")
+@Examples({
+	"if 5 is divisible by 5:",
+	"if 11 cannot be divided by 10:",
+})
+@Since("INSERT VERSION")
+public class CondIsDivisibleBy extends Condition {
+
+	static {
+		Skript.registerCondition(CondIsDivisibleBy.class,
+			"%numbers% (is|are) evenly divisible by %number%",
+			"%numbers% (isn't|is not|aren't|are not) evenly divisible by %number%",
+			"%numbers% can be evenly divided by %number%",
+			"%numbers% (can't|cannot|can not) be evenly divided by %number%");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<Number> divisorExpression;
+	@SuppressWarnings("null")
+	private Expression<Number> dividendExpression;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		divisorExpression = (Expression<Number>) exprs[0];
+		dividendExpression = (Expression<Number>) exprs[1];
+		setNegated(matchedPattern == 1 || matchedPattern == 3);
+		return true;
+	}
+
+	@Override
+	public boolean check(Event event) {
+		Number divisorNumber = divisorExpression.getSingle(event);
+		return divisorExpression.check(event, new Checker<Number>() {
+			@Override
+			public boolean check(Number dividendNumber) {
+				double dividend = dividendNumber.doubleValue();
+				double divisor = divisorNumber != null ? divisorNumber.doubleValue() : null;
```

Comment:
If divisorNumber is null, you're still trying to unbox it into a double, which will throw a NullPointerException.
Essentially, regardless of whether or not you do the null check, the `doubleValue()` method is still being called!